File: wrappers/matlab/depth_sensor.m
% Wraps librealsense2 depth_sensor class
classdef depth_sensor < realsense.sensor
    methods
        % Constructor
        function this = depth_sensor(handle)
            this = this@realsense.sensor(handle);
        end
        
        % Destructor (uses base class destructor)

        % Functions
        function scale = get_depth_scale(this)
            scale = realsense.librealsense_mex('rs2::depth_sensor', 'get_depth_scale', this.objectHandle);
        end
    end
end

File: wrappers/android/librealsense/src/main/java/com/intel/realsense/librealsense/DepthSensor.java
package com.intel.realsense.librealsense;

public class DepthSensor extends Sensor {

    DepthSensor(long handle) {
        super(handle);
        mOwner = false;
    }

    public float getDepthScale()  { return nGetDepthScale(mHandle); }


    private static native float nGetDepthScale(long handle);
}

